### ISSUE TITLE
Bundling fails

### ISSUE THREAD

**Issue by triniwiz** [NativeScript/nativescript-dev-webpack#85]
I get the following whenever i try bundling my app funny thing it worked before 

> ERROR in Error encountered resolving symbol values statically. Could not resolve ui/transition relative to /Users/triniwiz/Desktop/yoloApp/node_modules/tns-core-modules/ui/transition/fade-transition.d.ts., resolving symbol FadeTransition in /Users/triniwiz/Desktop/yoloApp/node_modules/tns-core-modules/ui/transition/fade-transition.d.ts

**Comment by sis0k0** [comment]
Hey, there. What's the TypeScript version? Can you please share your package.json an webpack config, too?

**Comment by triniwiz** [comment]
I think it uses `2.2.1`
```json
{
  "description": "yoloApp",
  "license": "",
  "readme": "",
  "repository": "",
  "nativescript": {
    "id": "com.triniwiz.yoloApp",
    "tns-android": {
      "version": "2.5.0"
    },
    "tns-ios": {
      "version": "2.5.0"
    }
  },
  "dependencies": {
    "@angular/common": "2.4.3",
    "@angular/compiler": "2.4.3",
    "@angular/core": "2.4.3",
    "@angular/forms": "2.4.3",
    "@angular/http": "2.4.3",
    "@angular/platform-browser": "2.4.3",
    "@angular/platform-browser-dynamic": "2.4.3",
    "@angular/router": "3.4.3",
    "email-validator": "^1.0.7",
    "moment": "^2.15.1",
    "nativescript-accordion": "^2.1.1",
    "nativescript-angular": "^1.4.0",
    "nativescript-audio": "latest",
    "nativescript-checkbox": "^1.2.6",
    "nativescript-couchbase": "^1.0.11",
    "nativescript-email": "^1.3.4",
    "nativescript-fresco": "^1.0.14",
    "nativescript-html-handler": "file:nativescript-html-handler",
    "nativescript-iqkeyboardmanager": "^1.0.1",
    "nativescript-local-notifications": "^1.1.8",
    "nativescript-pager": "^2.2.2",
    "nativescript-permissions": "^1.2.2",
    "nativescript-telerik-ui": "^1.6.1",
    "nativescript-toasty": "^1.1.1",
    "nativescript-tooltip": "^2.1.0",
    "nativescript-zip": "^1.0.0",
    "ng2-translate": "^3.1.2",
    "reflect-metadata": "^0.1.8",
    "rxjs": "~5.0.1",
    "tns-core-modules": "2.5.0",
    "tns-platform-declarations": "^2.5.0"
  },
  "devDependencies": {
    "@angular/compiler-cli": "2.4.3",
    "@ngtools/webpack": "1.2.4",
    "babel-traverse": "6.15.0",
    "babel-types": "6.15.0",
    "babylon": "6.11.2",
    "copy-webpack-plugin": "~3.0.1",
    "extract-text-webpack-plugin": "~2.0.0-beta.4",
    "htmlparser2": "~3.9.2",
    "lazy": "1.0.11",
    "nativescript-css-loader": "~0.26.0",
    "nativescript-dev-sass": "^0.4.1",
    "nativescript-dev-typescript": "^0.3.2",
    "nativescript-dev-webpack": "^0.3.3",
    "raw-loader": "~0.5.1",
    "resolve-url-loader": "~1.6.0",
    "typescript": "^2.0.3",
    "webpack": "2.2.0",
    "webpack-sources": "~0.1.3",
    "worker-loader": "git+https://github.com/triniwiz/worker-loader.git",
    "zone.js": "^0.7.2"
  },
  "scripts": {
    "clean-android": "tns clean-app android",
    "clean-ios": "tns clean-app ios",
    "prewebpack-android": "npm run clean-android",
    "prewebpack-ios": "npm run clean-ios",
    "webpack-android": "webpack --config=webpack.android.js --progress",
    "webpack-ios": "webpack --config=webpack.ios.js --progress",
    "prestart-android-bundle": "npm run webpack-android",
    "prestart-ios-bundle": "npm run webpack-ios",
    "start-android-bundle": "tns run android --bundle --disable-npm-install",
    "start-ios-bundle": "tns run ios --bundle --disable-npm-install",
    "prebuild-android-bundle": "npm run webpack-android",
    "prebuild-ios-bundle": "npm run webpack-ios",
    "build-android-bundle": "tns build android --bundle --disable-npm-install",
    "build-ios-bundle": "tns build ios --bundle --disable-npm-install",
    "rebuild": "rm -rf platforms/ && rm -rf node_modules && npm i"
  }
}
```

```js
var webpack = require("webpack");
var nsWebpack = require("nativescript-dev-webpack");
var nativescriptTarget = require("nativescript-dev-webpack/nativescript-target");
var path = require("path");
var CopyWebpackPlugin = require("copy-webpack-plugin");
var ExtractTextPlugin = require("extract-text-webpack-plugin");
var AotPlugin = require("@ngtools/webpack").AotPlugin;

module.exports = function (platform, destinationApp) {
    if (!destinationApp) {
        //Default destination inside platforms/<platform>/...
        destinationApp = nsWebpack.getAppPath(platform);
    }
    var entry = {};
    //Discover entry module from package.json
    entry.bundle = "./" + nsWebpack.getEntryModule();
    //Vendor entry with third party libraries.
    entry.vendor = "./vendor";
    //app.css bundle
    entry["app.css"] = "./app.css";

    var plugins = [
        new ExtractTextPlugin("app.css"),
        //Vendor libs go to the vendor.js chunk
        new webpack.optimize.CommonsChunkPlugin({
            name: ["vendor"]
        }),
        //Define useful constants like TNS_WEBPACK
        new webpack.DefinePlugin({
            "global.TNS_WEBPACK": "true",
        }),
        //Copy assets to out dir. Add your own globs as needed.
        new CopyWebpackPlugin([
            { from: "app.css" },
            { from: "css/**" },
            { from: "fonts/**" },
            { from: "**/*.jpg" },
            { from: "**/*.png" },
            { from: "**/*.xml" },
        ], { ignore: ["App_Resources/**"] }),
        //Generate a bundle starter script and activate it in package.json
        new nsWebpack.GenerateBundleStarterPlugin([
            "./vendor",
            "./bundle",
        ]),

        // Exclude explicitly required but never declared in XML elements. 
        // Loader nativescript-dev-webpack/tns-xml-loader should be added for *.xml/html and *.ts files.
        new nsWebpack.ExcludeUnusedElementsPlugin(),

        //Angular AOT compiler
        new AotPlugin({
            tsConfigPath: "tsconfig.aot.json",
            entryModule: path.resolve(__dirname, "app/app.module#AppModule"),
            typeChecking: false
        }),
        new nsWebpack.StyleUrlResolvePlugin({ platform }),
    ];

    if (process.env.npm_config_uglify) {
        plugins.push(new webpack.LoaderOptionsPlugin({
            minimize: true
        }));

        //Work around an Android issue by setting compress = false
        var compress = platform !== "android";
        plugins.push(new webpack.optimize.UglifyJsPlugin({
            mangle: {
                except: nsWebpack.uglifyMangleExcludes,
            },
            compress: compress,
        }));
    }

    return {
        context: path.resolve("./app"),
        target: nativescriptTarget,
        entry: entry,
        output: {
            pathinfo: true,
            path: path.resolve(destinationApp),
            libraryTarget: "commonjs2",
            filename: "[name].js",
        },
        resolve: {
            //Resolve platform-specific modules like module.android.js
            extensions: [
                ".aot.ts",
                ".ts",
                ".js",
                ".css",
                "." + platform + ".ts",
                "." + platform + ".js",
                "." + platform + ".css",
            ],
            //Resolve {N} system modules from tns-core-modules
            modules: [
                "node_modules/tns-core-modules",
                "node_modules"
            ]
        },
        node: {
            //Disable node shims that conflict with NativeScript
            "http": false,
            "timers": false,
            "setImmediate": false,
        },
        module: {
            loaders: [
                //Workers
                {
                    test: /\.worker.js$/,
                    loaders: [
                        "worker-loader"
                    ]
                },
                {
                    test: /\.html$|\.xml$/,
                    loaders: [
                        "raw-loader",
                        "nativescript-dev-webpack/tns-xml-loader",
                    ]
                },
                // Root app.css file gets extracted with bundled dependencies
                {
                    test: /app\.css$/,
                    loader: ExtractTextPlugin.extract([
                        "resolve-url-loader",
                        "nativescript-css-loader",
                        "nativescript-dev-webpack/platform-css-loader",
                    ]),
                },
                // Other CSS files get bundled using the raw loader
                {
                    test: /\.css$/,
                    exclude: /app\.css$/,
                    loaders: [
                        "raw-loader",
                    ]
                },
                // Compile TypeScript files with ahead-of-time compiler.
                {
                    test: /\.ts$/,
                    loaders: [
                        "nativescript-dev-webpack/tns-xml-loader",
                        "nativescript-dev-webpack/tns-aot-loader",
                        "@ngtools/webpack",
                    ]
                },
                // SASS support
                {
                    test: /\.scss$/,
                    loaders: [
                        "raw-loader",
                        "resolve-url-loader",
                        "sass-loader"
                    ]
                },
            ]
        },
        plugins: plugins,
    };
};

```

**Comment by triniwiz** [comment]
changing `"typescript": "^2.0.3"` => `"typescript": "~2.1.0"` fixed the problem `2.2.x` breaks things
